libraries/AP_Math/quaternion.cpp

```diff
@@ -424,34 +424,47 @@ void Quaternion::from_vector312(float roll, float pitch, float yaw)
 }
 
 // create a quaternion from its axis-angle representation
-void Quaternion::from_axis_angle(Vector3f v)
+void Quaternion::from_axis_angle(const Vector3f &v)
 {
-    const float theta = v.length();
-    if (is_zero(theta)) {
-        q1 = 1.0f;
-        q2=q3=q4=0.0f;
-        return;
+    const float theta_sq = v.x * v.x + v.y * v.y + v.z * v.z;
+    float re_coeff, im_coeff;
+    if (theta_sq < radians(0.1f) * radians(0.1f)) {
+        // When theta is close to 0, cannot divide v by theta to get the rotation axis
+        // Use Taylor series approximation of `sin(theta/2)/theta` and cos(theta/2) with second-order expansion
+        re_coeff = 1.0f - 0.125f * theta_sq;
```

Comment:
With changes to allow more double precision in parts of the code we've been asked not to use `1.0f` and instead `1.0` as the compiler can then decide between single vs. double precision.

```suggestion
        re_coeff = 1.0 - 0.125 * theta_sq;
```